osh/word_parse.py

```diff
@@ -667,7 +667,7 @@ def ReadSingleQuoted(self, lex_mode, left_token, tokens, is_ysh_expr):
 
                 # x = $'\z' is disallowed; ditto for echo $'\z' if shopt -u parse_backslash
                 if is_ysh_expr or not self.parse_opts.parse_backslash():
-                    p_die("Invalid char escape in C-style string literal", tok)
+                    p_die('Invalid char escape in C-style string literal. Use \\\\ to get \ (parse_backslash).', tok)
```

Comment:
Hm this seems overly specific, because the example given above is `\z`

I think what we could do is have some kind of error code, and the error code could explain the hints like using `\\\\` 


Though I don't even think that's a good workround, because mostly you will use regular single quotes like `'\'`

Reply:
Providing context and hints right with the error is immensely helpful. Needs to be provided directly, I think.

What do you think of adding them separately underneath? For example:
```
'Invalid char escape in C-style string literal (parse_backslash).
    Example options:          * \\\\ denotes \
                       posix: * '-string
                       ysh:   * r'-string
```

Reply:
```
'Invalid char escape in C-style string literal (parse_backslash).
\t* \\\\ to denote \ ?
\t* '-string ? (posix)
\t* r'-string ? (ysh)'
```

Reply:
> I don't even think that's a good workround, because mostly you will use regular single quotes like '\'

Workaround for what?
And to the contrary, if a $'...' string is used (explicit shell "raw" string supporting escape sequences), single quotes won't allow that.

Reply:
Please have a look at the changed strings.

Seems much better now. Adding "bullet questions" below the error could become a good general theme.